src/bgw/timer.c

```diff
@@ -48,8 +48,10 @@ get_timeout_millisec(TimestampTz by_time)
 
 	if (TIMESTAMP_IS_NOBEGIN(by_time))
 		return 0;
-
-	TimestampDifference(GetCurrentTimestamp(), by_time, &timeout_sec, &timeout_usec);
+	/* TimestampDifference takes arguments of type long which is 32bits on arm architectures.
+	 * This may cause issues if timeout_sec exceeds 32bits
+	 */
+	TimestampDifference(GetCurrentTimestamp(), by_time, (long *) &timeout_sec, &timeout_usec);
```

Comment:
why do we cast to (long*)? Could you explain why we need a cast here?
int64 is appropriately defined in  postgres include files (c.h).

Reply:
`TimestampDifference` expects a long* as an argument. on ARM, int64 is 64 bits while long is 32 bits and when we let the cast be done automatically, there is a compiler warning which would break travis testing.

Reply:
ok, on ARM, the type for int64 is actually long long int.  Lets fix this in a slightly different way.. a) Change type for  timeout_sec => long timeout_sec
b) cast the return value to int64.
57     return (int64)(timeout_sec * MILLISECS_PER_SEC + timeout_usec / USECS_PER_MILLI    SEC);